tests/conftest.py

```diff
@@ -30,6 +31,29 @@
 from piquasso._simulators.connectors import NumpyConnector
 
 
+
+def pytest_ignore_collect(path, config):
+    # Only apply this skip logic for Python >= 3.13
+    if sys.version_info >= (3, 13):
+        # Convert path to string for easier checks
+        path_str = str(path)
+
+        # Skip tests that are explicitly TensorFlow-related
+        if "tensorflow" in path_str.lower():
+            return True
+        
+        # Skip test files that are known to import TensorFlow
+        tf_related_files = [
+            "test_simulator_equivalence.py",
+            "test_tf_function.py",
+            "test_clements.py",
+            "test_gates.py",
+            "test_decompositions.py",
+        ]
+        return any(filename in path_str for filename in tf_related_files)
```

Comment:
So there is an issue with this, which is that it skips entire files, skipping tests also that are not using tensorflow.

For example, in the test `tests/_simulators/fock/test_gates.py::test_squeezing_probabilities`. We would still like to have it run for all the possible `SimulatorClass`es except for the ones that use `tensorflow`.

My first idea here is to the tensorflow import in a fixture, and when using Python 3.13, disable tests using this fixture. I don't know if this can be directly achieved in pytest, but I'm sure there is a solution.